### PR TITLE
Merge with upstream

### PR DESCRIPTION
**Pull request by IulianCrudu**
Fetch the changes applied on the upstream repo.

### CODE REVIEW

File: easyaudit/crudhistory_admin_mixin.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+from urllib.parse import urlencode
```
Comment (pcraciunoiu) [inline easyaudit/crudhistory_admin_mixin.py:1]: Any changes to this file, or is it identical to upstream?

**Review by pcraciunoiu — COMMENTED** [review]
These changes look good to me!

It would be good to try and run the tests now, and update the workflows to reduce to:

* Python 3.9, 3.10
* Django 3.2 (and maybe 4, if the tests run quickly)

File: easyaudit/crudhistory_admin_mixin.py (hunk 1 above)
Comment (IulianCrudu) [reply easyaudit/crudhistory_admin_mixin.py:1]: It's identical

**Comment by IulianCrudu** [comment]
> These changes look good to me!
> 
> It would be good to try and run the tests now, and update the workflows to reduce to:
> 
> * Python 3.9, 3.10
> * Django 3.2 (and maybe 4, if the tests run quickly)

Updated the workflows and had to fix a test, but now everything passes. I'll base the next branch(where I add the new feature) off of this one

File: easyaudit/signals/model_signals.py [hunk 2]
```diff
@@ -254,23 +258,22 @@ def m2m_changed(sender, instance, action, reverse, model, pk_set, using, **kwarg
...
                         changed_fields = []
                     else:
-                        changed_fields = {get_m2m_field_name(model, instance): list(pk_set)}
+                        pks = pk_set
+                        if isinstance(pks, Iterable):
+                            pks = (format_primary_key(pk) for pk in pks)
+                        else:
+                            pks = format_primary_key(pks)
+
+                        changed_fields = {get_m2m_field_name(model, instance): list(pks)}
```
Comment (pcraciunoiu) [inline easyaudit/signals/model_signals.py:276]: How come we don't need to do the `json.dumps` like upstream?

https://github.com/soynatan/django-easy-audit/blob/12d9d8b8b1d7dd627336c3fb8e609d5627c18966/easyaudit/signals/model_signals.py#L267

**Review by pcraciunoiu — COMMENTED** [review]
How come our version is 1.6.2? So far ahead of upstream.

Here is a comparison of the latest changes:

https://github.com/soynatan/django-easy-audit/compare/12d9d8b8b1d7dd627336c3fb8e609d5627c18966...UpliftAgency:django-easy-audit:merge-with-upstream

Could we create a PR for the JSONField conversion against the fork? Maybe they will review and upstream it. Seems fairly harmless.

Similarly, should we PR this session config too? https://github.com/UpliftAgency/django-easy-audit/commit/34e73fd18507b741fcf3026669df1339a2da933f

And the `get_model_queryset` setting... and others too.

Maybe start with the smallest PR?

File: easyaudit/signals/model_signals.py (hunk 2 above)
Comment (IulianCrudu) [reply easyaudit/signals/model_signals.py:276]: They use a TextField, but we use a JSONField

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -9,8 +9,8 @@ jobs:
     strategy:
       max-parallel: 4
       matrix:
-        python-version: [3.5, 3.6, 3.7, 3.8, 3.9, '3.10']
-        django-version: [2.2, 2.2.8, 3.0, 3.1, 3.2, 4.0]
+        python-version: [3.9, '3.10']
+        django-version: [3.2, 4.0]
```
Comment (camflan) [inline .github/workflows/build.yml:13]: do we need 4 here? I guess it doesn't hurt?

**Review by camflan — APPROVED** [review]

**Comment by pcraciunoiu** [comment]
Yeah let's PR this change?

**Review by pcraciunoiu — APPROVED** [review]
OK to merge

**Pull request merged**